tests/integration_tests/security/test_seccomp_validate.py

```diff
@@ -0,0 +1,134 @@
+# SPDX-License-Identifier: Apache-2.0
+
+"""Test that validates that seccompiler filters work as expected"""
+
+import json
+import platform
+import struct
+from pathlib import Path
+
+import pytest
+import seccomp
+
+from framework import utils
+from host_tools import cargo_build
+
+ARCH = platform.machine()
+
+
+@pytest.fixture(scope="session")
+def bin_test_syscall(test_fc_session_root_path):
+    """Build the test_syscall binary."""
+    test_syscall_bin = Path(test_fc_session_root_path) / "test_syscall"
+    cargo_build.gcc_compile("host_tools/test_syscalls.c", test_syscall_bin)
+    assert test_syscall_bin.exists()
+    yield test_syscall_bin
+
+
+class BpfMapReader:
+    """
+    Simple reader for the files that seccompiler-bin produces
+
+    The files are serialized with bincode[1] in format that is easy to parse.
+
+        sock_filter = <ushort uchar uchar uint>
+        BpfProgram = Vec<sock_filter>
+        BpfMap = BTreeMap(str, BpfProgram)
+        str = Vec<uchar>
+
+    [1] https://github.com/bincode-org/bincode/blob/trunk/docs/spec.md
+    """
+
+    INSN_FMT = "<HBBI"
+    INSN_SIZEOF = struct.calcsize(INSN_FMT)
+
+    def __init__(self, buf):
+        self.buf = buf
+        self.offset = 0
+
+    @classmethod
+    def from_file(cls, file):
+        """Initialize a buffer from a file"""
+        return cls(Path(file).read_bytes())
+
+    def read_format(self, fmt):
+        """Read a struct format string from the buffer"""
+        val = struct.unpack_from(fmt, self.buf, offset=self.offset)
+        self.offset += struct.calcsize(fmt)
+        if len(val) == 1:
+            return val[0]
+        return val
+
+    def is_eof(self):
+        """Are we at the end of the buffer?"""
+        return self.offset == len(self.buf)
+
+    def lookahead(self, size):
+        """Look ahead `size` bytes"""
+        return self.buf[self.offset : self.offset + size]
+
+    def split(self):
+        """Return separate filters"""
+        threads = {}
+        # how many entries in the map
+        map_len = self.read_format("<Q")
+        for _ in range(map_len):
+            # read key
+            key_str_len = self.read_format("<Q")
+            key_str = self.read_format(f"{key_str_len}s")
+            # read value: vec of instructions
+            insn_len = self.read_format("<Q")
+            data = self.lookahead(insn_len * self.INSN_SIZEOF)
+            threads[key_str.decode("ascii")] = data
+            self.offset += len(data)
+
+        assert self.is_eof()
+        return threads
+
+
+def test_validate_filter(seccompiler, bin_test_syscall, monkeypatch, tmp_path):
+    """Assert that the seccomp filter matches the JSON description."""
+
+    fc_filter_path = Path(f"../resources/seccomp/{ARCH}-unknown-linux-musl.json")
+    fc_filter = json.loads(fc_filter_path.read_text(encoding="ascii"))
+
+    # cd to a tmp dir because we are going to generate lots of coredumps
```

Comment:
are we actually interested in those coredumps? Can we just ulimit them to 0?

Reply:
I initially tried to find a command that would inhibit but found nothing. After your prompt I found Python's `resource.setrlimit`

Reply:
Fixed